--- MPlayer-1.0rc1/stream/udp.c.orig	2006-10-23 07:32:25.000000000 +0900
+++ MPlayer-1.0rc1/stream/udp.c	2007-02-25 00:28:27.000000000 +0900
@@ -98,6 +98,15 @@
   server_address.sin_family = AF_INET;
   server_address.sin_port = htons (url->port);
 
+  if ((ntohl (server_address.sin_addr.s_addr) >> 28) == 0xe)
+  {
+    unsigned int yes = 1;
+    if (setsockopt(socket_server_fd, SOL_SOCKET, SO_REUSEADDR,
+			    &yes, sizeof(yes))) {
+      mp_msg(MSGT_NETWORK, MSGL_ERR, "SO_REUSEADDR failed! ignore.\n");
+    }
+  }
+
   if (bind (socket_server_fd, (struct sockaddr *) &server_address,
             sizeof (server_address)) == -1)
   {
